### PR TITLE
fix: fatal log removed

### LINKED ISSUE #5054: Bug: Remove log.Fatal
### 📜 Description

log.Fatal causing microservices to restart

### 👟 Reproduction steps

Run a flow which has log.Fatal

### 👍 Expected behavior

Microservice must not restart. Just proper handeling must be done

### 👎 Actual Behavior

Microservices restarting

### ☸ Kubernetes version

AKS

### Cloud provider

<details>
Azure
</details>

### 🌍 Browser

Chrome

### 🧱 Your Environment

_No response_

### ✅ Proposed Solution

_No response_

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find any similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/devtron-labs/devtron/blob/main/CODE_OF_CONDUCT.md)

AB#9694

### PR DESCRIPTION
**Pull request by ShashwatDadhich**
# Description

log.fatal removed 

Fixes #5054

## How Has This Been Tested?

- [x] Tested locally 
- [x] Tested by deploying in VM.

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] Does this PR requires documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [x] I have tested it for all user roles.
* [x] I have added all the required unit/api test cases.

### CODE REVIEW

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: client/argocdServer/connection/proxy.go [hunk 1]
```diff
@@ -45,7 +45,10 @@ const (
 func NewCDHTTPReverseProxy(serverAddr string, transport http.RoundTripper, userVerifier func(token string) (int32, error)) func(writer http.ResponseWriter, request *http.Request) {
 	target, err := url.Parse(serverAddr)
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
```
Comment (nishant-d) [inline client/argocdServer/connection/proxy.go:48]: propagate error at startup

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: pkg/module/ModuleCacheService.go [hunk 2]
```diff
@@ -109,15 +111,17 @@ func (impl *ModuleCacheServiceImpl) updateModuleToInstalled(moduleName string) {
...
+		return
 	}
 }
 
 func (impl *ModuleCacheServiceImpl) buildInformerToListenOnInstallerObject() {
 	impl.logger.Debug("building informer cache to listen on installer object")
 	_, _, clusterDynamicClient, err := impl.K8sUtil.GetK8sInClusterConfigAndDynamicClients()
 	if err != nil {
-		log.Fatalln("not able to get k8s cluster rest config.", "error", err)
+		log.Println("not able to get k8s cluster rest config.", "error", err)
```
[the remark this answers is not in the document]
Comment (ShashwatDadhich) [reply pkg/module/ModuleCacheService.go:123]: done

File: client/argocdServer/connection/proxy.go (hunk 1 above)
Comment (ShashwatDadhich) [reply client/argocdServer/connection/proxy.go]: done

File: pkg/auth/authorisation/casbin/Adapter.go [hunk 3]
```diff
@@ -61,21 +62,24 @@ func Create() *casbin.SyncedEnforcer {
...
 	}
 	auth, err1 := casbin.NewSyncedEnforcerSafe("./auth_model.conf", a)
 	if err1 != nil {
-		log.Fatal(err1)
+		log.Println(err1)
+		return nil, err
 	}
 	e = auth
 	err = e.LoadPolicy()
 	log.Println("casbin Policies Loaded Successfully")
```
[the remark this answers is not in the document]
Comment (ShashwatDadhich) [reply pkg/auth/authorisation/casbin/Adapter.go]: done

File: fetchAllEnv/fetchAllEnv.go [hunk 4]
```diff
@@ -88,7 +88,7 @@ func processGoFile(filePath string, allFields *[]EnvField, uniqueKeys *map[strin
 	fset := token.NewFileSet()
 	node, err := parser.ParseFile(fset, filePath, nil, parser.ParseComments)
 	if err != nil {
-		log.Fatalln("error parsing file:", err)
+		log.Println("error parsing file:", err)
```
[the remark this answers is not in the document]
Comment (ShashwatDadhich) [reply fetchAllEnv/fetchAllEnv.go:98]: done

File: client/grafana/proxy.go [hunk 5]
```diff
@@ -7,15 +7,18 @@ import (
 	"net/url"
 )
 
-func NewGrafanaHTTPReverseProxy(serverAddr string, transport http.RoundTripper) func(writer http.ResponseWriter, request *http.Request) {
+func NewGrafanaHTTPReverseProxy(serverAddr string, transport http.RoundTripper) (func(writer http.ResponseWriter, request *http.Request), error) {
 	target, err := url.Parse(serverAddr)
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
+		return func(writer http.ResponseWriter, request *http.Request) {
```
Comment (nishant-d) [inline client/grafana/proxy.go]: remove error function

Comment (nishant-d) [reply client/grafana/proxy.go]: true for all

File: pkg/module/ModuleCacheService.go [hunk 6]
```diff
@@ -109,15 +111,18 @@ func (impl *ModuleCacheServiceImpl) updateModuleToInstalled(moduleName string) {
 	}
 	err := impl.moduleRepository.Save(module)
 	if err != nil {
-		log.Fatalln("Error while saving module.", "moduleName", moduleName, "error", err)
+		log.Println("Error while saving module.", "moduleName", moduleName, "error", err)
```
Comment (nishant-d) [inline pkg/module/ModuleCacheService.go:114]: return error

File: pkg/server/ServerCacheService.go [hunk 7]
```diff
@@ -74,11 +75,12 @@ func NewServerCacheServiceImpl(logger *zap.SugaredLogger, serverEnvConfig *serve
 		// fetch current version from helm release
 		releaseInfo, err := impl.helmAppService.GetValuesYaml(context.Background(), &appIdentifier)
 		if err != nil {
-			log.Fatalln("got error in fetching devtron helm release values.", "error", err)
+			log.Println("got error in fetching devtron helm release values.", "error", err)
```
Comment (nishant-d) [inline pkg/server/ServerCacheService.go:78]: return error

**Review by nishant-d — CHANGES_REQUESTED** [review]

**Review by nishant-d — DISMISSED** [review]

**Review by nishant-d — APPROVED** [review]

**Pull request merged**